File: bumperbot_ws/src/bumperbot_cpp_examples/src/simple_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

#include <string>

#include <chrono>
using std::placeholders::_1;

class SimpleSubscriber : public rclcpp::Node
{
private:
    //const std::string node_name  = "simple_subscriber";
    const std::string topic_name = "chatter";
    const int que_size = 10;
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;

    void msgCallback(const std_msgs::msg::String &msg) const
    {
        RCLCPP_INFO_STREAM(get_logger(), "I heard:" << msg.data.c_str());
    }

public:
    SimpleSubscriber() : Node("simple_subscriber")
    {
        sub_ = create_subscription<std_msgs::msg::String>(topic_name, que_size, std::bind(&SimpleSubscriber::msgCallback, this, _1));
    }
};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleSubscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: bumperbot_ws/src/bumperbot_py_examples/bumperbot_py_examples/simple_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimpleSubscriber(Node):
    def __init__(self):
        # Publisherの変数
        node_name  = "simple_subscriber"
        topic_name = "chatter" 
        que_size   = 10
                
        super().__init__(node_name)
        
        self.sub_ = self.create_subscription(String, topic_name, self.msgCallback, que_size)
    
    def msgCallback(self, msg):
        self.get_logger().info("I heard: %s" % msg.data)

def main():
    rclpy.init()
    simple_subscriber = SimpleSubscriber()
    rclpy.spin(simple_subscriber)
    simple_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()
        